tests/forms_tests/field_tests/test_urlfield.py

```diff
@@ -121,17 +129,46 @@ def test_urlfield_clean_required(self):
         with self.assertRaisesMessage(ValidationError, msg):
             f.clean("")
 
+    @ignore_warnings(category=RemovedInDjango60Warning)
     def test_urlfield_clean_not_required(self):
         f = URLField(required=False)
         self.assertEqual(f.clean(None), "")
         self.assertEqual(f.clean(""), "")
 
+    @ignore_warnings(category=RemovedInDjango60Warning)
     def test_urlfield_strip_on_none_value(self):
         f = URLField(required=False, empty_value=None)
         self.assertIsNone(f.clean(""))
         self.assertIsNone(f.clean(None))
 
+    @ignore_warnings(category=RemovedInDjango60Warning)
     def test_urlfield_unable_to_set_strip_kwarg(self):
         msg = "__init__() got multiple values for keyword argument 'strip'"
         with self.assertRaisesMessage(TypeError, msg):
             URLField(strip=False)
+
+    @ignore_warnings(category=RemovedInDjango60Warning)
+    def test_urlfield_assume_scheme(self):
+        f = URLField()
+        self.assertEqual(f.clean("example.com"), "http://example.com")
+        f = URLField(assume_scheme="http")
+        self.assertEqual(f.clean("example.com"), "http://example.com")
+        f = URLField(assume_scheme="https")
+        self.assertEqual(f.clean("example.com"), "https://example.com")
+
+    def test_urlfield_raises_warning(self):
+        msg = (
+            "The URLField assumes 'http' if no scheme is specified. "
+            "From Django 6.0, the URLField will assume 'https'. "
+            "A new keyword argument 'assume_scheme' has been introduced. "
+            "To retain the previous behavior, use URLField(assume_scheme='http')."
+        )
+        with self.assertWarnsMessage(RemovedInDjango60Warning, msg):
+            URLField()
+
+    def test_urlfield_with_assume_scheme_no_warning(self):
+        with warnings.catch_warnings():
+            URLField(assume_scheme="http")
+
+        with warnings.catch_warnings():
+            URLField(assume_scheme="https")
```

Comment:
All warning are treated as errors in Django test suite so we can reorganize it a bit:
```suggestion
    def test_urlfield_assume_scheme(self):
        with ignore_warnings(category=RemovedInDjango60Warning):
            f = URLField()
            self.assertEqual(f.clean("example.com"), "https://example.com")
        f = URLField(assume_scheme="http")
        self.assertEqual(f.clean("example.com"), "http://example.com")
        f = URLField(assume_scheme="https")
        self.assertEqual(f.clean("example.com"), "https://example.com")
    
    def test_urlfield_raises_warning(self):
        msg = (
            "The URLField assumes 'http' if no scheme is specified. "
            "From Django 6.0, the URLField will assume 'https'. "
            "A new keyword argument 'assume_scheme' has been introduced. "
            "To retain the previous behavior, use URLField(assume_scheme='http')."
        )
        with self.assertWarnsMessage(RemovedInDjango60Warning, msg):
            f = URLField()
            self.assertEqual(f.clean("example.com"), "http://example.com")
```